src/lib/components/SpiderChart.svelte

```diff
@@ -0,0 +1,214 @@
+	// `tickToPolygon` draws the concentric octagons
+	function tickToPolygon(tick: number) {
+		const numbers = features
+			.map((f, i) => {
+				let angle = Math.PI / 2 + (2 * Math.PI * i) / features.length;
+				return angleToCoordinate(angle, tick);
+			})
+			.reduce((prev, curr) => {
+				return `${prev} ${curr.x} ${curr.y}`;
+			}, '')
+			.trim();
+		return numbers;
+	}
+
+	// `radialTickLines` calculates the lines from center of the octagons to create the web
+	function radialTickLines() {
+		let lines: { outerX: number; outerY: number }[] = [];
+		for (var i = 0; i < features.length; i++) {
+			let pct = i / features.length;
+			let angle = Math.PI / 2 + 2 * Math.PI * pct;
+			const { x, y } = angleToCoordinate(angle, 5);
+			lines.push({
+				outerX: x,
+				outerY: y
+			});
+		}
+		return lines;
+	}
+
+	function drawAnswerShape(answers: Record<string, number>): string | null {
+		let coordinates: [number, number][] = [];
+		for (var i = 0; i < features.length; i++) {
+			let ft_name = features[i];
+			let angle = Math.PI / 2 + (2 * Math.PI * i) / features.length;
+			let { x, y } = angleToCoordinate(angle, answers[ft_name]);
+			coordinates.push([x, y]);
+		}
+		return lineHelper(coordinates);
+	}
+
+	// draws circles for score labels
+	function getCircleCoords({ answer, idx }: { answer: number; idx: number }) {
+		let angle = Math.PI / 2 + (2 * Math.PI * idx) / features.length;
+		return {
+			...angleToCoordinate(angle, answer),
+			answer
+		};
+	}
+	const formattedAnswers = $derived(
+		features.reduce(
+			(
+				resultArray: {
+					feature: string;
+					answer: number;
+					xCoord: number;
+					yCoord: number;
+					idx: number;
+				}[],
+				feature,
+				idx
+			) => {
+				const answer = answers[feature];
+				const coords = getCircleCoords({ answer, idx });
+				resultArray.push({ feature, answer, xCoord: coords.x, yCoord: coords.y, idx });
+				return resultArray;
+			},
+			[]
+		)
+	);
+
+	// calculates the offset of each label's centerpoint from the center of the chart
+	let labels = $derived.by(() => {
+		const numPoints = features.length;
+		const points = [];
+		const angleStep = (2 * Math.PI) / numPoints;
+		// not using radialScale because we want a radius outside those bounds
+		const radius = config.d / 2 + config.p + config.labelD / 2;
+
+		for (let i = 0; i < numPoints; i++) {
+			const angle = i * angleStep;
+			const x = radius * Math.sin(angle);
+			// multiply y by -1 so it maps clockwise
+			const y = radius * Math.cos(angle) * -1;
+			points.push({ offsetX: x, offsetY: y, text: dynamics[i] });
+		}
+		return points;
+	});
+</script>
+
+<svelte:window bind:innerWidth />
+<div class="outer" style={`padding: ${config.labelD}px;`}>
+	<svg id="chart" width={config.d + 2 * config.p} height={config.d + 2 * config.p}>
+		<g id="ticks">
+			{#each config.ticks as tick}
+				<!-- concentric octogons -->
+				<polygon points={tickToPolygon(tick)} />
+			{/each}
+			{#each radialTickLines() as f}
```

Comment:
😮 cool that the return of `radialTickLines` doesn't have to stored in a named var before being used